rotkehlchen/api/rest.py

```diff
@@ -5009,3 +5010,58 @@ def get_event_details(self, identifier: int) -> Response:
             return api_response(wrap_in_fail_result('No details found'), status_code=HTTPStatus.NOT_FOUND)  # noqa: E501
 
         return api_response(_wrap_in_ok_result(details), status_code=HTTPStatus.OK)
+
+    def add_evm_transaction_hash(
+            self,
+            async_query: bool,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> Response:
+        if async_query is True:
+            return self._query_async(
+                command=self._add_evm_transaction_hash,
+                evm_chain=evm_chain,
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+
+        response = self._add_evm_transaction_hash(
+            evm_chain=evm_chain,
+            tx_hash=tx_hash,
+            associated_address=associated_address,
+        )
+        status_code = _get_status_code_from_async_response(response)
+        result_dict = {'result': response['result'], 'message': response['message']}
+        return api_response(process_result(result_dict), status_code=status_code)
+
+    def _add_evm_transaction_hash(
+            self,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> dict[str, Any]:
+        evm_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain)
+        try:
+            transaction, tx_receipt = evm_manager.transactions.add_transaction_hash(
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+        except (KeyError, DeserializationError, RemoteError) as e:
+            return wrap_in_fail_result(
+                message=(
+                    f'unable to add transaction with hash {tx_hash.hex()} and '
+                    f'associated address {associated_address} due to {str(e)}'
+                ),
```

Comment:
You are duplicating the error message in both branches. You can just have a single except clause and do an 

`isinstance(e, sqlcipher.IntegrityError)` to decide what error code to use